tests/Crypto/tst_cryptorequests/tst_cryptorequests.cpp

```diff
@@ -280,125 +264,131 @@ void tst_cryptorequests::getPluginInfo()
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsLocking, 0);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsSetLockCode, 0);
-        } else if (p.name() == TEST_USB_TOKEN_PLUGIN_NAME) {
-            foundTestUsbPlugin = true;
-            QCOMPARE(p.statusFlags() & PluginInfo::Available, PluginInfo::Available);
-            QCOMPARE(p.statusFlags() & PluginInfo::MasterUnlocked, PluginInfo::MasterUnlocked);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginUnlocked, 0);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsLocking, PluginInfo::PluginSupportsLocking);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsSetLockCode, 0);
+
+    if (pluginType == CryptoTest::CryptoPlugin) {
+        QCOMPARE(cryptoPluginsChangedSpy.count(), 1);
+        bool foundPlugin = false;
+        for (auto p : r.cryptoPlugins()) {
+            if (p.name() == pluginName) {
+                foundPlugin = true;
+                QCOMPARE(p.displayName(), pluginDisplayName);
+                QCOMPARE(p.statusFlags(), pluginStatus);
+                break;
+            }
         }
+        QVERIFY(foundPlugin);
     }
-    for (auto p : r.storagePlugins()) {
-        if (p.name() == DEFAULT_TEST_CRYPTO_STORAGE_PLUGIN_NAME) {
-            foundTestCryptoStoragePluginInS = true;
-            QCOMPARE(p.displayName(), QStringLiteral("SQLCipher"));
-            QCOMPARE(p.statusFlags() & PluginInfo::Available, PluginInfo::Available);
-            QCOMPARE(p.statusFlags() & PluginInfo::MasterUnlocked, PluginInfo::MasterUnlocked);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginUnlocked, PluginInfo::PluginUnlocked);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsLocking, 0);
-            QCOMPARE(p.statusFlags() & PluginInfo::PluginSupportsSetLockCode, 0);
+
+    if (pluginType == CryptoTest::StoragePlugin) {
+        QCOMPARE(storagePluginsChangedSpy.count(), 1);
+        bool foundPlugin = false;
+        for (auto p : r.storagePlugins()) {
+            if (p.name() == pluginName) {
+                foundPlugin = true;
+                QCOMPARE(p.displayName(), pluginDisplayName);
+                QCOMPARE(p.statusFlags(), pluginStatus);
+                break;
+            }
         }
+        QVERIFY(foundPlugin);
     }
+}
 
-    QCOMPARE(foundTestCryptoPlugin, true);
-    QCOMPARE(foundTestCryptoStoragePluginInC, true);
-    QCOMPARE(foundTestCryptoStoragePluginInS, true);
-    QCOMPARE(foundTestUsbPlugin, true);
+void tst_cryptorequests::randomData_data()
+{
+    QTest::addColumn<TestPluginMap>("plugins");
+    QTest::addColumn<QString>("csprngEngineName");
+    QTest::addColumn<int>("generateRandomDataSize");    // zero if plugin does not support GenerateRandomDataRequest
+    QTest::addColumn<double>("seedRandomDataEntropyEstimate");
+    QTest::addColumn<QByteArray>("seedData");   // empty if plugin does not support SeedRandomDataGeneratorRequest
+
```

Comment:
I guess we should also add a column like "expectedResult" which would be a Result(Result::Succeeded) if the plugin should support this operation, or Result(Result::OperationNotSupportedError, QString()) for plugins which don't support it.  The actual test should compare just the result.code() and result.errorCode() I guess.